### DISCUSSION TITLE
What is the configuration detection principle of Kubescape? Is it scanning the yaml file?

### DISCUSSION THREAD

**Discussion by testwill** [kubescape/kubescape#1194] [Q&A] [answered]
What is the configuration detection principle of Kubescape? Is it scanning the yaml file?

**Comment by matthyx** [comment, marked as answer]
Hi @testwill it depends on what you want to achieve, Kubescape can scan the following:
- local file (YAML, Helm chart, ...)
- local dir containing Kubernetes resource definitions
- local git repo (check out)
- remote git repo
- Kubernetes clusters (via api access)

If you want to scan a single YAML file, please invoke kubescape like:
```
kubescape scan file.yaml
```
